tuf/client/updater.py

```diff
@@ -1811,9 +1839,6 @@ def _update_metadata(self, metadata_role, upperbound_filelength, version=None):
     updated_metadata_object = metadata_signable['signed']
     current_metadata_object = self.metadata['current'].get(metadata_role)
 
-    self._verify_root_chain_link(metadata_role, current_metadata_object,
```

Comment:
Why did we remove this code again?

Reply:
It's just duplicating steps that have already been performed in `_verify_metadata_file` and not performing those steps until after the tempfile has been persisted.